crates/ruff_python_formatter/src/expression/expr_if_exp.rs

```diff
@@ -53,9 +55,25 @@ impl FormatNodeRule<ExprIfExp> for FormatExprIfExp {
         let comments = f.context().comments().clone();
 
         let inner = format_with(|f: &mut PyFormatter| {
-            // We place `if test` and `else orelse` on a single line, so the `test` and `orelse` leading
-            // comments go on the line before the `if` or `else` instead of directly ahead `test` or
-            // `orelse`
+            // If the expression has any leading or trailing comments, and is "alone" in brackets,
+            // always expand it, as in:
+            // ```
+            // [
+            //     # comment
+            //     0
+            //     if self.thing is None
+            //     else before - after
+            // ]
+            // ```
+            if (comments.has_leading(item) || comments.has_trailing(item))
+                && is_expression_bracketed(item.into(), f.context().source())
```

Comment:
We already track whether we are inside of parentheses in `NodeLevel`

```suggestion
                && f.context().node_level().is_parenthesized()
```

Reply:
I don’t think this is quite the same — the condition here is “parenthesized or the only item in a set of brackets (e.g., the only item in a tuple literal or list literal or call).”